ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -0,0 +1,706 @@
+            For departure date the key "From" will be set to True.
+        """
+        date_dict_list = []
+        regex_pattern = re.compile(r'\b((onward date\:|onward date -|departure date|leaving on|starting from|'
+                                   r'departing on|departing|going on|for|departs on)\s+(.+))\b')
+        patterns = regex_pattern.findall(self.processed_text.lower())
+
+        for pattern in patterns:
+            date_dict_list.extend(
+                self._date_dict_from_text(text=pattern[2], from_property=True)
+            )
+        return date_dict_list
+
+    def _detect_return_date(self):
+        """
+        Finds return type dates in the given text by matching few keywords like 'coming back', 'return date',
+        'leaving on', 'returning on', 'returning at', 'arriving', 'arrive' . It detects dates in the part of text right
+        to these keywords.
+
+        Args:
+
+        Returns:
+            The list of dictionary containing the dictionary for date which is detected as return date.
+            For departure date the key "to" will be set to True.
+        """
+
+        date_dict_list = []
+        regex_pattern = re.compile(r'\s?((coming back|back|return date\:?|return date -|returning on|'
+                                   r'arriving|arrive|return|returning at)\s+(.+))\.?\s?')
+        patterns = regex_pattern.findall(self.processed_text.lower())
+        for pattern in patterns:
+            date_dict_list.extend(
+                self._date_dict_from_text(text=pattern[2], to_property=True)
+            )
+        return date_dict_list
+
+    def _detect_any_date(self):
+        """
+        Finds departure and return type dates in the given text. It detects 'departure' and 'return' and their synonyms
+        and tags all the detected dates in the text accordingly. If both type synonyms are found, and more than one
+        dates are detected, first date is marked as departure type and last as return type. If only one date is found,
+        it is marked as departure if 'departure' or both type synonyms are found and marked as 'return' type if
+        'return' type synonyms were found in the given text
+
+
+        Args:
+
+        Returns:
+            The list of dictionary containing the dictionary for date which is detected as departure date and date
+            that got detected as arrival date or the date that got detected as normal date. For departure date
+            the key "from" will be set to True.
+            Whereas for arrival date the key "to" will be set to True.
+            Otherwise the normal date with the key 'normal' will be set to True
+        """
+        date_dict_list = []
+        departure_date_flag = False
+        return_date_flag = False
+        if self.bot_message:
+            departure_regex_string = r'traveling on|going on|starting on|departure date|date of travel|' + \
```

Comment:
1) Hindi date time ner changes missing from here.
2) These functions of departure and return date are not language agnostic. Please put these constants(leaving on, starting from, etc) in respective lang folder Like hi, gu, etc. so they scale for languages as well.